src/main/scala/org/allenai/pipeline/s3/S3Artifact.scala

```diff
@@ -171,7 +183,7 @@ trait CompressedS3Artifact extends S3Artifact {
   }
 
   override def readContents(): InputStream =
-    new BufferedInputStream(new BZip2CompressorInputStream(super.readContents()))
+    new BZip2CompressorInputStream(super.readContents())
```

Comment:
This `BufferedInputStream` was always on the wrong side of `BZip2CompressorInputStream` (it has to always be on the filesystem side). Now we don't need it at all anymore, because `super.readContents()` returns a `BufferedInputStream` already.